packages/deployment-server-ui/src/pages/TypeId.tsx

```diff
@@ -1,68 +1,117 @@
-import { DeployRequest, DeviceType, Device } from "deployment-server";
-import React, { useEffect, useState } from "react";
+import { DeployRequest, Device, DeviceTypeResource } from "deployment-server";
+import { useEffect, useState } from "react";
 import defineFunctionalComponent from "../util/defineFunctionalComponent";
-import { Card, Col, Row } from "antd";
+import { Card, Col, Row, Spin, Statistic } from "antd";
 import { useInterval } from "react-use";
 import { useParams } from "react-router-dom";
 import { StatusTag } from "../components/StatusTag";
+import DeploymentsStatusGraph from "../components/Graphs/DeploymentsStatusGraph";
+import IssueCountGraph from "../components/Graphs/IssueCountGraph";
+import Meta from "antd/lib/card/Meta";
 
 export default defineFunctionalComponent(function TypeId() {
-  let [devices, setDevices] = useState<Device[]>([]);
-  let [deviceType, setDeviceType] = useState<DeviceType | null>(null);
-  let [deployments, setDeployments] = useState<DeployRequest[]>([]);
-  let [refetchFlip, setRefetchFlip] = useState(false);
+  const [loading, setLoading] = useState<boolean>(true);
+  const [devices, setDevices] = useState<Device[]>([]);
+  const [deviceType, setDeviceType] = useState<DeviceTypeResource>();
+  const [deployments, setDeployments] = useState<DeployRequest[]>([]);
+  const [refetchFlip, setRefetchFlip] = useState(false);
 
-  const { id } = useParams()
+  const { id } = useParams();
   function findDeviceById(id: string) {
-    return devices.find((device) => id === device.id)
+    return devices.find((device) => id === device.id);
   }
+
   useInterval(function () {
     setRefetchFlip(!refetchFlip);
-  }, 5000);
+  }, 1000);
 
   useEffect(() => {
-    fetch(`/api/device-types/${id}`).then(async (res) => {
-      setDeviceType(await res.json());
-    });
+    const fetchAsync = async () => {
+      const typesRes = await fetch(`/api/device-types/${id}`);
+      const types = await typesRes.json();
+      setDeviceType(types);
+
+      const devRes = await fetch(`/api/devices`);
+      const devs = await devRes.json();
+      setDevices(devs.filter((dev: Device) => dev.deviceTypeId === id));
+
+      const deployRes = await fetch("/api/deployments");
+      const deploys = await deployRes.json();
+      setDeployments(
+        deploys.filter(
+          // @ts-ignore: Unreachable code error
+          (deployment) =>
+            deployment.deviceId === findDeviceById(deployment.deviceId)?.id
+        )
+      );
+    };
+
+    fetchAsync();
   }, []);
 
   useEffect(() => {
-    fetch(`/api/devices`)
-      .then(async (res) => {
-        const devices: Device[] = await res.json();
-        return devices;
-      })
-      .then((res) => {
-        setDevices(
-          res.filter((device) => device.deviceTypeId === id)
-        );
-      });
-  }, []);
+    if (
+      devices != null &&
+      deviceType != null &&
+      deployments != null &&
+      loading
```

Comment:
`&& loading ==(=) false` (?)

Does not make sense why we would set Loading to false is `loading` is truthy

Reply:
That part is from me. The reason why I did this is so the the loading parameter isn't being set every time anything in devices, deviceTypes or deployments changes. It's not like this makes the code much more efficient but I still wanted to avoid this unnecessary state change (or set to same state again)...